### PR TITLE
feat: Adding validation to Cloudformation to reject same template_body

### PR DESCRIPTION
**Pull request by teddylear**
feat: Adding validation to Cloudformation to reject same template_body and parameters

First PR to moto, please let me know if I'm missing anything or if anything should be changed. 

Addresses #4157

### CODE REVIEW

File: tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py [hunk 1]
```diff
@@ -1023,6 +1021,35 @@ def test_boto3_update_stack_fail_missing_new_parameter():
...
+            StackName=name,
+            TemplateBody=dummy_template_yaml_with_ref,
+            Parameters=params,
+        )
+    exp_err = exp.value.response.get("Error")
+    exp_metadata = exp.value.response.get("ResponseMetadata")
+
+    exp_err.get("Code").should.match(r"ValidationError")
+    print(f"Message recieved: {exp_err.get('Message')}")
+    assert exp_err.get("Message") == f"Stack [{name}] already exists"
```
Comment (bblommers) [inline tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py:1049]:
```suggestion
    exp_err.get("Code").should.equal("ValidationError")
    exp_err.get("Message").should.equal(f"Stack [{name}] already exists")
```
This is more readable IMO

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @teddylear, thanks for the PR!
Would you mind extending the test, and call the `update_stack`-method with updated parameters at the end?
There's no need to verify that the update-method itself works as expected, that hasn't changed - just to verify that the error will not be thrown with different params.

File: tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py (hunk 1 above)
Comment (teddylear) [reply tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py]: Agreed, updated to match this.

**Comment by teddylear** [comment]
@bblommers 
Updated the test to call update stack with new parameters. Please let me know if there's anything else I should update.

**Review by bblommers — APPROVED** [review]
LGTM! Thanks for the PR @teddylear

**Pull request merged**